README.md

```diff
@@ -0,0 +1,491 @@
+
+```javascript
+import { BaseError } from 'path-to-error-handler';
+
+throw new BaseError('Custom error message', 404, { additionalInfo: 'details' });
+```
+
+## Invalid Path Handler
+
+The ```invalidPathHandler``` middleware function handles undefined paths, returning a 404 error.
+
+### Usage
+
+```javascript
+import { invalidPathHandler } from 'path-to-error-handler';
+
+// ... Other middleware and route handling
+
+app.use(invalidPathHandler);
+```
+
+## Password Validation
+
+A list of common passwords is available for use in password validation.
+
+### Usage
+
+```javascript
+import { commonPasswords100 } from 'path-to-error-handler';
+
+// Example of password validation
+const userPassword = 'password123';
+if (commonPasswords100.includes(userPassword)) {
+  // Password is common, handle accordingly
+}
+```
+
+### Common Passwords List
+
+The ```commonPasswords100``` array contains commonly used passwords and can be used for password strength validation.
+
+## Notes
+
+- Ensure that the error handling middleware functions are appropriately placed in the middleware stack to catch errors.
+- Custom errors can be used for more specific error handling with detailed status codes and error messages.
+
+Feel free to customize and expand upon this documentation based on your specific application requirements.
+
+# Authentication Middleware Documentation
+
+The authentication middleware in this application utilizes JSON Web Tokens (JWT) to verify and extract user information from incoming requests.
+
+## Overview
+
+The authentication middleware is designed to secure specific routes by ensuring that requests contain a valid JWT token in the 'Authorization' header. If the token is valid, the middleware decodes it, extracts the user ID, and attaches it to the request object for further use in downstream middleware or route handlers.
+
+## Usage
+
+The `auth` middleware function should be applied to routes or groups of routes that require authentication.
+
+### Example
+
+```javascript
+import { auth, CustomRequest } from 'path-to-authentication-middleware';
+
+// ... Other middleware and route handling
+
+// Apply authentication middleware to a route
+app.get('/authenticated-route', auth, (req: CustomRequest, res) => {
+  // Access user ID from the decoded token
+  const userId = req.token;
+
+  // ... Other route handling logic
+});
+```
+
+## Middleware Function
+
+
+The `auth` middleware function performs the following steps:
+
+1. Extracts the JWT token from the 'Authorization' header.
+2. Verifies the token's validity using the provided secret key.
+3. Decodes the token to retrieve the user ID.
+4. Attaches the user ID to the request object as `req.token`.
+5. Calls the next middleware or route handler if authentication is successful.
+
+### Custom Request Interface
+
+The `CustomRequest` interface extends the Express `Request` interface to include the `token` property.
```

Comment:
Don't think this should be documented here. This should be in the relevant code file instead of ReadMe. The ReadMe should be a guide on how to USE the software, not WHY the software does what it does.

Reply:
good point, i went from showing the code and then to explain some of the code.